test/SparqlParserTest.cpp

```diff
@@ -1179,3 +1180,95 @@ TEST(ParserTest, Bind) {
+        pq._rootGraphPattern._children[0].variant_));
+  }
+  {
+    ParsedQuery pq =
+        SparqlParser("SELECT ?x ?y WHERE { ?x :myrel ?y } ORDER BY ASC(?y)")
+            .parse();
+    ASSERT_EQ(pq._orderBy.size(), 1);
+    EXPECT_THAT(pq._orderBy[0], IsVariableOrderKey("?y", false));
+    ASSERT_EQ(pq._rootGraphPattern._children.size(), 1);
+    ASSERT_TRUE(holds_alternative<GraphPatternOperation::BasicGraphPattern>(
+        pq._rootGraphPattern._children[0].variant_));
+  }
+  {
+    ParsedQuery pq =
+        SparqlParser("SELECT ?x ?y WHERE { ?x :myrel ?y } ORDER BY DESC(?foo)")
+            .parse();
+    ASSERT_EQ(pq._orderBy.size(), 1);
+    EXPECT_THAT(pq._orderBy[0], IsVariableOrderKey("?foo", true));
+    ASSERT_EQ(pq._rootGraphPattern._children.size(), 1);
+    ASSERT_TRUE(holds_alternative<GraphPatternOperation::BasicGraphPattern>(
+        pq._rootGraphPattern._children[0].variant_));
+  }
+  {
+    ParsedQuery pq =
+        SparqlParser(
+            "SELECT ?x WHERE { ?x :myrel ?y } GROUP BY ?x ORDER BY "
+            "?x")
+            .parse();
+    ASSERT_EQ(pq._orderBy.size(), 1);
+    EXPECT_THAT(pq._orderBy[0], IsVariableOrderKey("?x", false));
+    ASSERT_EQ(pq._rootGraphPattern._children.size(), 1);
+    ASSERT_TRUE(holds_alternative<GraphPatternOperation::BasicGraphPattern>(
+        pq._rootGraphPattern._children[0].variant_));
+  }
+  {
+    EXPECT_THROW(
+        SparqlParser("SELECT ?x WHERE { ?x :myrel ?y } GROUP BY ?x ORDER BY "
+                     "?y")
+            .parse(),
+        ParseException);
+  }
+  {
+    ParsedQuery pq =
+        SparqlParser(
+            "SELECT ?x (COUNT(?y) as ?c) WHERE { ?x :myrel ?y } GROUP "
+            "BY ?x ORDER BY ?c")
+            .parse();
+    ASSERT_EQ(pq._orderBy.size(), 1);
+    EXPECT_THAT(pq._orderBy[0], IsVariableOrderKey("?c", false));
+    ASSERT_EQ(pq._rootGraphPattern._children.size(), 1);
+    ASSERT_TRUE(holds_alternative<GraphPatternOperation::BasicGraphPattern>(
+        pq._rootGraphPattern._children[0].variant_));
+  }
+  {
+    ParsedQuery pq =
+        SparqlParser("SELECT ?x ?y WHERE { ?x :myrel ?y } ORDER BY (?x - ?y)")
+            .parse();
+    ASSERT_EQ(pq._orderBy.size(), 1);
+    ASSERT_EQ(pq._rootGraphPattern._children.size(), 2);
+    ASSERT_TRUE(holds_alternative<GraphPatternOperation::BasicGraphPattern>(
+        pq._rootGraphPattern._children[0].variant_));
+    auto variant = pq._rootGraphPattern._children[1].variant_;
+    ASSERT_TRUE(holds_alternative<GraphPatternOperation::Bind>(variant));
+    auto helperBind = get<GraphPatternOperation::Bind>(variant);
+    ASSERT_EQ(helperBind._expression.getDescriptor(), "?x-?y");
+    ASSERT_EQ(pq._orderBy[0].variable_, helperBind._target);
+  }
+  {
+    EXPECT_THROW(
+        SparqlParser("SELECT ?y WHERE { ?x :myrel ?y } GROUP BY ?y ORDER BY "
+                     "(?x - ?y)")
+            .parse(),
+        ParseException);
+  }
```

Comment:
You should have at least two error cases:
One that contains a no-grouped variable from the body (always illegal, `ORDER BY ?x` in your case)
One that contains grouped variables (`ORDER BY (2 * ?y)`) but in an expression (not supported by QLever).

Otherwise one of the conditions is never checked during the tests.
Also those tests should have a short description, what is wrong here (the tests where the parsing works are pretty straightforward).

Reply:
The first case is already here:  https://github.com/ad-freiburg/qlever/blob/6abe627fd20d161531e0c13dc545b01b1986130c/test/SparqlParserTest.cpp#L1234-L1240
I'll add an additional test case with grouping and an expression that only contains grouped variables.